src/dataEntryApp/sagas/programEncounterSaga.js

```diff
@@ -223,7 +264,78 @@ export function* loadEditProgramEncounterWorker({ programEncounterUuid, enrolUui
   programEnrolment.enrolmentDateTime = new Date(programEnrolmentJson.enrolmentDateTime);
   programEncounter.programEnrolment = programEnrolment;
 
-  yield put(setProgramEncounterForm(mapForm(programEncounterForm)));
+  yield put.resolve(setProgramEncounterForm(mapForm(programEncounterForm)));
+  yield put.resolve(setProgramEncounter(programEncounter));
+  yield put.resolve(getSubjectProfile(programEnrolmentJson.subjectUuid));
+  yield put.resolve(setLoad(true));
+}
+
+export function* cancelProgramEncounterFetchFormWatcher() {
+  yield takeLatest(types.GET_CANCEL_PROGRAM_ENCOUNTER_FORM, cancelProgramEncounterFetchFormWorker);
+}
+
+export function* cancelProgramEncounterFetchFormWorker({ programEncounterUuid, enrolmentUuid }) {
+  yield put.resolve(setLoad(false));
+  yield put.resolve(setCancelProgramEncounterForm());
+  const programEncounterJson = yield call(api.fetchProgramEncounter, programEncounterUuid);
+  const programEnrolmentJson = yield call(api.fetchProgramEnrolment, enrolmentUuid);
+  const subjectProfileJson = yield call(api.fetchSubjectProfile, programEnrolmentJson.subjectUuid);
+  yield put.resolve(setSubjectProfile(mapProfile(subjectProfileJson)));
+
+  const subjectProfile = yield select(state => state.dataEntry.subjectProfile.subjectProfile);
+  const formMapping = yield select(
+    selectCancelProgramEncounterFormMapping(
+      programEncounterJson.encounterType.uuid,
+      programEnrolmentJson.program.uuid,
+      subjectProfile.subjectType.uuid
+    )
+  );
+
+  const cancelProgramEncounterForm = yield call(api.fetchForm, formMapping.formUUID);
+  const programEncounter = mapProgramEncounter(programEncounterJson);
+  programEncounter.cancelDateTime = new Date();
+  programEncounter.cancelObservations = [];
+  const programEnrolment = new ProgramEnrolment();
+  programEnrolment.uuid = enrolmentUuid;
+  programEnrolment.enrolmentDateTime = new Date(programEnrolmentJson.enrolmentDateTime);
+  programEncounter.programEnrolment = programEnrolment;
+
+  yield put.resolve(setCancelProgramEncounterForm(mapForm(cancelProgramEncounterForm)));
```

Comment:
No need to dispatch three different actions. We can combine them all in a single action. Action name might be something like cancelFormLoaded and it can take form and programEncoutner as parameters. And It can also set the load flag true inside the reducer to be true.